container-images/tcib/base/os/tempest/run_tempest.sh

```diff
@@ -499,10 +499,10 @@ if [ ${TEMPEST_DEBUG_MODE} == true ]; then
     sleep infinity
 fi
 
-if [ -s ${TEMPEST_LOGS_DIR}stestr_failing.txt ] && [ -s ${TEMPEST_EXPECTED_FAILURES_LIST} ]; then
+if [ -n "${TEMPEST_EXPECTED_FAILURES_LIST}" ] && [ -s ${TEMPEST_LOGS_DIR}stestr_failing.txt ] && [ -s "${TEMPEST_EXPECTED_FAILURES_LIST}" ]; then
```

Comment:
So, the problem with that variable is that is not defined at all in this file ~ it is expected to come from test-operator as environment variable https://github.com/openstack-k8s-operators/test-operator/blob/main/controllers/tempest_controller.go#L432

I think I would just ensure the default value of some /dev/null – to make it also more clear that this variable is not containing a list/string but a path to file...

Reply:
With this change, if it the variable is not set e.g `[ -n "${var}}" ]` is evaluated to FALSE - it skips the entire block. Which would be the right thing to do when no "flaky" tests are set.

Setting it to `/dev/null` by default is less efficient, because then it needs to do IO to check the file exist and is not empty.

Reply:
> Did you really just raise IO-related efficiency argument in Bash script...? 😆

Yes I did. :laughing:  I am :100: for pervasively optimizing even in shell scripts.

Reply:
Ok, jokes aside – I would still much more like to have default value defined, instead of adding another condition on variable in place were it may be simply unset. For me, this script should have `nounset` option enabled by default – and I will be aiming for that – hence I am more in favor of solution introduced as part of https://github.com/openstack-k8s-operators/tcib/pull/320

Reply:
Also, are we prepared to immidiatly backport the refactoring in #320 ?

Reply:
So, I was just deeply testing it with my team this morning, we are about to merge and backport ASAP

Reply:
> So, I was just deeply testing it with my team this morning, we are about to merge and backport ASAP

ok, let's leave this open. If yours merge soon, feel free to close this one. If there are delays, I think we should do this one - re-basing your re-factoring should not be that painful ... :)